Integrations/integration-Tanium.yml

```diff
@@ -0,0 +1,483 @@
+      description: Tanium id of object (to be used instead of name)
+    - name: object_type
+      required: true
+      auto: PREDEFINED
+      predefined:
+      - user
+      - package
+      - saved_question
+      - sensor
+      - saved_action
+      description: Type of object to get
+    description: Send a generic Get Object request
+  - name: tn-get-all-saved-questions
+    arguments: []
+    outputs:
+    - contextPath: Tanium.SavedQuestions.query_text
+      description: Question query text
+    - contextPath: Tanium.SavedQuestions.mod_time
+      description: Question modification time
+    - contextPath: Tanium.SavedQuestions.user.id
+      description: Unique id of user who saved question
+    - contextPath: Tanium.SavedQuestions.user.name
+      description: Name of user who saved question
+    - contextPath: Tanium.SavedQuestions.name
+      description: Name of saved question
+    - contextPath: Tanium.SavedQuestions.expire_seconds
+      description: Question expire time in seconds
+    - contextPath: Tanium.SavedQuestions.id
+      description: Saved question unique id
+    - contextPath: Tanium.SavedQuestions.issue_seconds
+      description: Issue time in seconds
+    description: Gets all saved questions
+  - name: tn-deploy-package
+    arguments:
+    - name: package
+      required: true
+      description: Name of package to deploy with this action
+    - name: action_filters
+      description: Comma separated list of strings. Each string must describe a sensor
+        and a filter which limits which computers the action will deploy package to
+    - name: action_options
+      description: Comma separated list of options to apply to action_filters
+    - name: get_results
+      auto: PREDEFINED
+      predefined:
+      - "True"
+      - "False"
+      description: Specifies whether ait for result completion after deploying action
+    - name: package_args
+      description: Comma separated list of arguments neede to execute the package
+        command
+    outputs:
+    - contextPath: Tanium.SavedActions.Id
+      description: Saved action id
+    - contextPath: Tanium.SavedActions.Name
+      description: Saved action name
+    description: Deploy a package and get the results back
+  - name: tn-ask-question
+    arguments:
+    - name: question
+      required: true
+      description: The question text
+    outputs:
+    - contextPath: Tanium.QuestionResults
+      description: Results of requested question. May be a complex object
+    description: Ask the server to parse the question text and choose the first parsed
+      result as the question to run
+  - name: tn-ask-system
+    arguments:
+    - name: hostname
+      description: Name of host
+    outputs:
+    - contextPath: Tanium.QuestionResults
+      description: Results of requested computer name. May be a complex object
+    description: Ask a question about a specific endpoint
+  - name: tn-get-saved-question
+    arguments:
+    - name: name
```

Comment:
I think that one of name and id arguments should be required. and one of them should be default. No?

Reply:
you need to use either one. if I set one to required you won't be able to use it without it.

Reply:
Ok, didn't know that